### ISSUE TITLE
ClientOptions sugar

### ISSUE THREAD

**Issue by mpenet** [lerouxrgd/clj-lettuce#2]
sugar for https://github.com/lettuce-io/lettuce-core/wiki/Client-options
This could be a simple multimethod that builds the options from a map

**Comment by lerouxrgd** [comment]
I deployed a 0.1.1-SNAPSHOT with the addition of `set-options` to `RedisConnector` protocol.
Does it work for you ?

Besides, I updated Lettuce to 4.3.3.Final because 4.4.0 made the unit tests failed (some geocommands), I still have to investigate it before a final release.

**Comment by mpenet** [comment]
Thanks! 

A nice pattern for wrapping builders is to use multimethods instead of a closed cond. It makes it extendable at runtime potentially, see ex here https://github.com/mpenet/spandex/blob/master/src/clj/qbits/spandex/client_options.clj (set-options). It might be opinionated but I found it works/reads well

**Comment by lerouxrgd** [comment]
Thanks, now I see what you meant with the multimethod approach.
I think it's good when you manipulate a complex builder (in Spandex you reify interfaces, reduce over options, etc).
The Lettuce builders I wrapped here are simpler I think, I feel it would be a bit overkill to use the multimethod pattern for them.
If `client-options` complexity grows I'll consider switching to multimethods.

Note that I might change things a bit (passing client-options at `RedisConnector` instantation time, etc).
I will wait for 4.4.1 to finalize this as the [gecommand issue was a bug](https://github.com/lettuce-io/lettuce-core/issues/593).

**Comment by mpenet** [comment]
Nice. I had modified my local version to take :client-options as an arg
already :) (I am on a trip, limited connection). I also made it so that
enums are passed via keywords (inferred/verified automatically from the
enums). Lastly I changed a few constructors to take an option map instead
of unrolled arguments, imho it composes better (no need to apply f when
it's dynamic). The same branch also has client options as a multimethod in
separate ns (like in alia/spandex).

I can create a few prs when I am on a decent connection if you d like.

Cheers!



On Aug 29, 2017 13:20, "Leroux Romain" <notifications@github.com> wrote:

Thanks, now I see what you meant with the multimethod approach.
I think it's good when you manipulate a complex builder (in Spandex you
reify interfaces, reduce over options, etc).
The Lettuce builders I wrapped here are simpler I think, I feel it would be
a bit overkill to use the multimethod pattern for them.
If client-options complexity grows I'll consider switching to multimethods.

Note that I might change things a bit (passing client-options at
RedisConnector instantation time, etc).
I will wait for 4.4.1 to finalize this as the gecommand issue was a bug
<https://github.com/lettuce-io/lettuce-core/issues/593>.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/lerouxrgd/celtuce/issues/2#issuecomment-325633577>, or mute
the thread
